launch/aloha_bringup.launch.py

```diff
@@ -13,596 +15,424 @@
     OpaqueFunction,
 )
 from launch.conditions import (
-  IfCondition,
+    IfCondition,
 )
 from launch.launch_description_sources import PythonLaunchDescriptionSource
 from launch.substitutions import (
     EnvironmentVariable,
     LaunchConfiguration,
     PathJoinSubstitution,
+    FindExecutable,
+    Command,
+    TextSubstitution,
 )
 from launch_ros.actions import Node
 from launch_ros.substitutions import FindPackageShare
 from launch_ros.parameter_descriptions import ParameterFile
 
 
+def load_yaml_file(yaml_path):
+    # Function to read and parse the YAML file
+    with open(yaml_path, "r") as f:
+        return yaml.safe_load(f)
+
+
 def launch_setup(context, *args, **kwargs):
-    robot_model_leader_launch_arg = LaunchConfiguration('robot_model_leader')
-    robot_model_follower_launch_arg = LaunchConfiguration('robot_model_follower')
 
-    robot_name_leader_left_launch_arg = LaunchConfiguration('robot_name_leader_left')
-    robot_name_leader_right_launch_arg = LaunchConfiguration('robot_name_leader_right')
-    robot_name_follower_left_launch_arg = LaunchConfiguration('robot_name_follower_left')
-    robot_name_follower_right_launch_arg = LaunchConfiguration('robot_name_follower_right')
+    nodes = []
 
-    leader_modes_left_launch_arg = LaunchConfiguration('leader_modes_left')
-    leader_modes_right_launch_arg = LaunchConfiguration('leader_modes_right')
-    follower_modes_left_launch_arg = LaunchConfiguration('follower_modes_left')
-    follower_modes_right_launch_arg = LaunchConfiguration('follower_modes_right')
+    # Get the base name from the 'robot_config_name' launch argument and construct the YAML path
+    robot_config_base = LaunchConfiguration("robot").perform(context)
+    yaml_file_path = PathJoinSubstitution(
+        [
+            FindPackageShare("aloha"),
+            "config",
+            "robot",
+            TextSubstitution(text=f"{robot_config_base}.yaml"),
+        ]
+    ).perform(context)
```

Comment:
Can do this to help with readability, reduce indentation

```suggestion
    yaml_file_path = PathJoinSubstitution([
        FindPackageShare("aloha"),
        "config",
        "robot",
        TextSubstitution(text=f"{robot_config_base}.yaml"),
    ]).perform(context)
```